tests/e2e-test-framework/conftest.py

```diff
@@ -97,13 +104,53 @@ def wire_mock():
         Config.requests_verify = False
         yield wire_mock
 
+def get_utils(request) -> Utils:
+    return Utils(
+        vm_user=request.config.getoption("--login"), 
+        vm_cred=request.config.getoption("--password"), 
+        namespace=request.config.getoption("--namespace")
+    )
+
+def get_ssh_executors(request) -> dict[str, SSHCommandExecutor]:
+    utils  = get_utils(request)
+    worker_ips = utils.get_worker_ips()
+    executors = {ip: SSHCommandExecutor(ip_address=ip, username=utils.vm_user, password=utils.vm_cred) for ip in worker_ips}
+    return executors
+
+@pytest.fixture(scope="session")
+def utils(request) -> Utils:
+    return get_utils(request)
+
+@pytest.fixture(scope="session")
+def ssh_executors(request) -> dict[str, SSHCommandExecutor]:
+    return get_ssh_executors(request)
+
+@pytest.fixture(scope="session")
+def drive_utils_executors(request) -> dict[str, DriveUtils]:
+    ssh_execs = get_ssh_executors(request)
+    return {ip: DriveUtils(executor) for ip, executor in ssh_execs.items()}
+
 @pytest.fixture(scope="function", autouse=True)
 def link_requirements_in_background(request):
     if pytest.qtest_helper is not None:
         requirements_thread = PropagatingThread(target=link_requirements, args=(request,), test_name=request.node.name)
         requirements_thread.start()
         pytest.threads.append(requirements_thread)
-    
+
+@pytest.fixture(autouse=True)
+def keep_drive_count(drive_utils_executors: dict[str, DriveUtils]) -> Generator[None, None, None]:
```

Comment:
to make sure we don't disconnect a drive during a test without plugging it again